core/providers/java/gradle.go

```diff
@@ -0,0 +1,105 @@
+package java
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/railwayapp/railpack/core/generate"
+)
+
+const (
+	DEFAULT_GRADLE_VERSION = 8
+	GRADLE_CACHE_KEY       = "gradle"
+)
+
+func (p *JavaProvider) usesGradle(ctx *generate.GenerateContext) bool {
+	return ctx.App.HasMatch("gradlew")
+}
+
+func (p *JavaProvider) getGradleVersion(ctx *generate.GenerateContext) (int, error) {
```

Comment:
Instead of just returning a version, this should set the version for gradle on the context. That allows us to have "source" info which is displayed to the user. Basically lets them know why a specific version was used. [Deafult versions](https://railpack.com/architecture/package-resolution#previous-and-default-versions) are also treated bit differently than versions from files.

https://github.com/railwayapp/railpack/blob/main/core/providers/node/node.go#L241

![screenshot-2025-03-05-13 57 56](https://github.com/user-attachments/assets/949d40e9-2392-404b-869b-f9793270f6e9)